cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -306,4 +333,41 @@ private void publishStopProgram(String stopPayload) {
     programStateWriter.killed(programRunId);
     LOG.debug("Published program stop message for run {}", programRunInfo.getRun());
   }
+
+  /**
+   * Fetch program status updates and filter if they are initiated by a tethered peer.
+   */
+  private Map<String, TetheringProgramUpdate> getPeerProgramUpdates() {
```

Comment:
looks like a Map never has multiple elements and is only being used to return a pair. In situations like this, it's better to include the peer name as an attribute in the TetheringProgramUpdate class and just return a single object, or return some other class that has the peer name and the TetheringProgramUpdate.

Though see above comment about re-using AbstractNotificationSubscriberService, which has logic for reading from the topic and decoding it, which could make this method not needed.